gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -56,19 +56,22 @@
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.ThreadSafe;
 
-/** Google Cloud Storage {@link FileSystemProvider} */
+/**
+ * Google Cloud Storage {@link FileSystemProvider}.
+ */
 @ThreadSafe
 @AutoService(FileSystemProvider.class)
 public final class CloudStorageFileSystemProvider extends FileSystemProvider {
 
-  private static final Supplier<GcsService> gcsServiceSupplier =
-      memoize(new Supplier<GcsService>() {
-        @Override
-        public GcsService get() {
-          return GcsServiceFactory.createGcsService();
-        }});
+  private final Storage storage;
 
-  private final GcsService gcsService;
+  // used only when we create a new instance of CloudStorageFileSystemProvider.
+  private static StorageOptions storageOptions;
+
+  /** Those options are only used by the CloudStorageFileSystemProvider ctor. */
+  public static void setGCloudOptions(StorageOptions newStorageOptions) {
```

Comment:
I find this method strange.

I think one could not assume when calling this method that it will
have an impact on the instance created by nio library and if one needs to create an instance
of this class with option we should just make that constructor public.